11_circular_csimeone/scr/plot_the_driest_and_wettest_days_CRB.R

```diff
@@ -0,0 +1,197 @@
+library(tibble)
+library(dplyr)
+library(tidyr)
+library(lubridate)
+library(data.table)
+library(readr)
+library(RColorBrewer)
+library(stringr)
+library(ggplot2)
+library(circular)
+library(cowplot)
+library("dataRetrieval")
+
+cfs_to_mm <- 28320000 * 86400 / (1000000 * 1000000)
+
+# Read in metadata for plotting. 
+# Metadata are derived from Gages II metadata here: https://water.usgs.gov/GIS/metadata/usgswrd/XML/gagesII_Sept2011.xml
+# and from site information from each site available on NWIS. 
+station_metadata <- read.csv("11_circular_csimeone/data_in/NWIS_data/CRB/colorado_metadata_all.csv",header=TRUE,stringsAsFactors = FALSE)
+station_metadata$site <- str_pad(station_metadata$site, width=8, side="left", pad = "0") 
+loc_df <- station_metadata %>%
+  as_tibble() %>%
+  rename(latitude = lat, longitude = long) %>%
+  filter(HUC02 %in% c(14, 15)) %>%
+  dplyr::select(c(site, HUC02, DRAIN_SQKM))
+
+load("11_circular_csimeone/data_in/site_list_CRB.RData")
+
+
+pCd <- "00060" #Discharge, cubic feet per second, see readNWISdv help file for more options
+start_date <- as.Date("1980-04-01")
+end_date <- as.Date("2021-12-31")
+
+# for (i in 1:length(site_list)){
+#   #NWIS data pull
+#   station <- site_list[i]
+#   cat(station, i, "of", length(site_list), "\n")
+# 
+#   df <- readNWISdv(station,pCd,start_date,end_date) %>%
+#     renameNWISColumns() %>%
+#     as_tibble()
+# 
+#   write_csv(df, paste0("11_circular_csimeone/data_in/NWIS_data/CRB/", station, ".csv"))
+# }
+
+# Read in all individual percentile files. 
+df_list <- c()
+for (i in 1:length(site_list)){
+  tryCatch({
+    print(i)
+    
+    df_temp <- read.csv(paste0("11_circular_csimeone/data_in/NWIS_data/CRB/", site_list[[i]], ".csv")) %>%
+      as_tibble()
+    # df_temp$site <- site_list[[i]]
+    
+    df_list[[i]] <- df_temp
+  }, error=function(e){cat("Site", i, "ERROR ", conditionMessage(e), "\n")})
+}
+
+df_ucrb <- bind_rows(df_list) %>%
```

Comment:
Include a comment to describe what is happening here

Reply:
Added comment for upper and lower basin